src/adapter/src/coord/sequencer/inner/create_materialized_view.rs

```diff
@@ -448,45 +469,14 @@ impl Coordinator {
                     cluster_id,
                     non_null_assertions,
                     custom_logical_compaction_window: compaction_window,
-                    refresh_schedule: refresh_schedule.clone(),
+                    refresh_schedule,
+                    initial_as_of: Some(as_of.clone()),
                 }),
                 owner_id: *ctx.session().current_role_id(),
             }),
         )
         .collect::<Vec<_>>();
 
-        // Timestamp selection
-        let as_of = {
-            // Normally, `as_of` should be the least_valid_read.
-            let id_bundle = dataflow_import_id_bundle(global_lir_plan.df_desc(), cluster_id);
-            let mut as_of = self.least_valid_read(&id_bundle);
-            // But for MVs with non-trivial REFRESH schedules, it's important to set the
-            // `as_of` to the first refresh. This is because we'd like queries on the MV to
-            // block until the first refresh (rather than to show an empty MV).
-            if let Some(refresh_schedule) = &refresh_schedule {
-                if let Some(as_of_ts) = as_of.as_option() {
-                    let Some(rounded_up_ts) = refresh_schedule.round_up_timestamp(*as_of_ts) else {
-                        return Err(AdapterError::MaterializedViewWouldNeverRefresh(
-                            refresh_schedule.last_refresh().expect("if round_up_timestamp returned None, then there should be a last refresh"),
-                            *as_of_ts
-                        ));
-                    };
-                    as_of = Antichain::from_elem(rounded_up_ts);
-                } else {
-                    // The `as_of` should never be empty, because then the MV would be unreadable.
-                    soft_panic_or_log!("creating a materialized view with an empty `as_of`");
-                }
-            }
-            as_of
-        };
-
-        // If we have a refresh schedule that has a last refresh, then set the `until` to the last refresh.
-        // (If the `try_step_forward` fails, then no need to set an `until`, because it's not possible to get any data
-        // beyond that last refresh time, because there are no times beyond that time.)
-        let until = refresh_schedule
-            .and_then(|s| s.last_refresh())
-            .and_then(|r| r.try_step_forward());
-
```

Comment:
I moved this part into `select_timestamps`, not because it was necessary but to clean up a bit. The logic inside `select_timestamps` is unchanged and only slightly reformatted.